presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -823,6 +823,32 @@ private void assertSTGeometryN(String wkt, int index, String expected)
         assertFunction("ST_ASText(ST_GeometryN(ST_GeometryFromText('" + wkt + "')," + index + "))", VARCHAR, expected);
     }
 
+    @Test
+    public void testSTGeometries()
+    {
+        assertSTGeometries("POINT EMPTY", null);
+        assertSTGeometries("LINESTRING (77.29 29.07, 77.42 29.26, 77.27 29.31, 77.29 29.07)", "LINESTRING (77.29 29.07, 77.42 29.26, 77.27 29.31, 77.29 29.07)");
+        assertSTGeometries("POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))");
+        assertSTGeometries("MULTIPOINT (1 2, 4 8, 16 32)", "POINT (1 2)", "POINT (4 8)", "POINT (16 32)");
+        assertSTGeometries("MULTILINESTRING ((1 1, 2 2))", "LINESTRING (1 1, 2 2)");
+        assertSTGeometries("MULTIPOLYGON (((0 0, 1 0, 1 1, 0 1, 0 0)), ((1 1, 3 1, 3 3, 1 3, 1 1)))",
+                "POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", "POLYGON ((1 1, 3 1, 3 3, 1 3, 1 1))");
+        assertSTGeometries("GEOMETRYCOLLECTION (POINT (2 3), LINESTRING (2 3, 3 4))", "POINT (2 3)", "LINESTRING (2 3, 3 4)");
+    }
+
+    private void assertSTGeometries(String wkt, String... expected)
+    {
+        ImmutableList<String> expectedList;
+        if (expected == null) {
+            expectedList = null;
+        }
+        else {
+            expectedList = ImmutableList.copyOf(expected);
+        }
```

Comment:
Since there is only one caller for the `null` value, it might be clearer to inline that caller and  not handle nulls here.

`assertFunction("ST_Geometries(ST_GeometryFromText('POINT EMPTY'))", new ArrayType(VARCHAR), null);`